app/models/team.rb

```diff
@@ -35,11 +35,28 @@ class Team < ApplicationRecord
   has_many :reports, inverse_of: :team, dependent: :destroy
   has_many :datatables_reports,
            class_name: 'Views::Datatables::DatatablesReport'
+  has_many :view_states, as: :viewable, dependent: :destroy
 
   after_commit do
     Views::Datatables::DatatablesReport.refresh_materialized_view
   end
 
+  def current_view_state(user)
```

Comment:
I'm just thinking, could we isolate this (albeit not very big) code block into its own service, maybe even an abstracted one (e.g. have an abstract service interface for retrieving - and possibly saving - state for any user - object combo)?

I'm generally not a fan of Ruby's fat models paradigm (after seeing how bloated things can become). I just feel that this various state-keeping logic for GUI components doesn't really belong scattered into multiple `ActiveRecord` data classes. If anything, I myself conceptually relate these states more to users than to their related objects. That's why I think isolated services might be the most clean solution.

What do you think?

Reply:
I can move it ot Concern